CMakeExt/HDF5.cmake

```diff
@@ -59,18 +54,33 @@ find_package_handle_standard_args(
   HDF5_INCLUDE_DIRS
 )
 
+
 # set flags
 if(HDF5_FOUND)
-#  set (HDF5_LINKER_FLAGS "-lhdf5_hl -lhdf5 -ldl -lm -lz")
-   set (HDF5_LINKER_FLAGS $ENV{HDF5_LIB} ${SZIP_LIB} -lz)
+  # save current state
+  cmake_push_check_state(RESET)
+  set(CMAKE_REQUIRED_INCLUDES ${HDF5_INCLUDE_DIRS})
+
+  CHECK_SYMBOL_EXISTS(H5_HAVE_PARALLEL "H5pubconf.h" HAVE_H5_PARALLEL)
+
+  if(NOT HAVE_H5_PARALLEL)
+    message(STATUS "HDF5 provides only serial version")
+    set(HDF5_FOUND OFF CACHE BOOL "HDF5_FOUND" FORCE)
+    unset(HDF5_LIBRARIES)
+    unset(HDF5_INCLUDE_DIRS)
+  else()
+#   set (HDF5_LINKER_FLAGS "-lhdf5_hl -lhdf5 -ldl -lm -lz")
+    set (HDF5_LINKER_FLAGS $ENV{HDF5_LIB} ${SZIP_LIB} -lz)
+  endif()
+  cmake_pop_check_state()
```

Comment:
I find it clearer, that the pop is directly after the `CHECK_SYMBOL_EXISTS`. Also the style is not consistent, either ALL_APPER_CASE or all_lower_case.